### PR TITLE
electrodialysis flowsheet improvement 

### PR DESCRIPTION
**Pull request by lbibl**
## Fixes/Resolves:

Improved optimization problem initialization for shorter solver iteration.  Added new variables to reduce testing errors. 

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/examples/flowsheets/electrodialysis/electrodialysis_1stack.py [hunk 1]
```diff
@@ -115,6 +117,45 @@ def build():
...
+    m.fs.eq_mem_area = Constraint(
+        expr=m.fs.mem_area
+        == m.fs.EDstack.cell_width
+        * m.fs.EDstack.cell_length
+        * m.fs.EDstack.cell_pair_num
+    )
+    # scaling constraint
+    iscale.constraint_scaling_transform(m.fs.eq_product_salinity, 1)
+    iscale.constraint_scaling_transform(m.fs.eq_disposal_salinity, 1)
+    iscale.constraint_scaling_transform(m.fs.eq_mem_area, 1)
```
Comment (bknueven) [inline watertap/examples/flowsheets/electrodialysis/electrodialysis_1stack.py]:
```suggestion
```
This doesn't seem necessary.

**Review by bknueven — APPROVED** [review]
We should merge this. For the `optimize` step of this flowsheet on main (current):
```
Number of Iterations....: 1462

                                   (scaled)                 (unscaled)
Objective...............:   2.4640040438785563e-01    2.4640040438785563e-01
Dual infeasibility......:   3.8573552088829352e-12    1.3921686442729345e-09
Constraint violation....:   4.2656232279031370e-13    9.6633812063373625e-13
Complementarity.........:   2.5059715319726150e-09    2.5059715319726150e-09
Overall NLP error.......:   3.0207529331274158e-10    2.5059715319726150e-09


Number of objective function evaluations             = 13921
Number of objective gradient evaluations             = 1458
Number of equality constraint evaluations            = 13921
Number of inequality constraint evaluations          = 0
Number of equality constraint Jacobian evaluations   = 1464
Number of inequality constraint Jacobian evaluations = 0
Number of Lagrangian Hessian evaluations             = 1462
Total CPU secs in IPOPT (w/o function evaluations)   =     15.530
Total CPU secs in NLP function evaluations           =      1.232
```

With this PR (new):
```
Number of Iterations....: 29

                                   (scaled)                 (unscaled)
Objective...............:   2.4640040438781419e-01    2.4640040438781419e-01
Dual infeasibility......:   1.3993701154156360e-12    5.0505050505535989e-10
Constraint violation....:   1.9455828500184597e-13    9.0949470177292824e-13
Complementarity.........:   9.0909090909091282e-10    9.0909090909091282e-10
Overall NLP error.......:   9.0909090909091282e-10    9.0909090909091282e-10


Number of objective function evaluations             = 33
Number of objective gradient evaluations             = 30
Number of equality constraint evaluations            = 33
Number of inequality constraint evaluations          = 0
Number of equality constraint Jacobian evaluations   = 30
Number of inequality constraint Jacobian evaluations = 0
Number of Lagrangian Hessian evaluations             = 29
Total CPU secs in IPOPT (w/o function evaluations)   =      0.318
Total CPU secs in NLP function evaluations           =      0.011
```

The solution is higher-quality and takes a reasonable number of iterations.

There may be some work to do on appropriate variable scaling, either on this flowsheet or the associated units.

**Comment by lbibl** [comment]
> We should merge this. For the `optimize` step of this flowsheet on main (current):
> 
> ```
> Number of Iterations....: 1462
> 
>                                    (scaled)                 (unscaled)
> Objective...............:   2.4640040438785563e-01    2.4640040438785563e-01
> Dual infeasibility......:   3.8573552088829352e-12    1.3921686442729345e-09
> Constraint violation....:   4.2656232279031370e-13    9.6633812063373625e-13
> Complementarity.........:   2.5059715319726150e-09    2.5059715319726150e-09
> Overall NLP error.......:   3.0207529331274158e-10    2.5059715319726150e-09
> 
> 
> Number of objective function evaluations             = 13921
> Number of objective gradient evaluations             = 1458
> Number of equality constraint evaluations            = 13921
> Number of inequality constraint evaluations          = 0
> Number of equality constraint Jacobian evaluations   = 1464
> Number of inequality constraint Jacobian evaluations = 0
> Number of Lagrangian Hessian evaluations             = 1462
> Total CPU secs in IPOPT (w/o function evaluations)   =     15.530
> Total CPU secs in NLP function evaluations           =      1.232
> ```
> 
> With this PR (new):
> 
> ```
> Number of Iterations....: 29
> 
>                                    (scaled)                 (unscaled)
> Objective...............:   2.4640040438781419e-01    2.4640040438781419e-01
> Dual infeasibility......:   1.3993701154156360e-12    5.0505050505535989e-10
> Constraint violation....:   1.9455828500184597e-13    9.0949470177292824e-13
> Complementarity.........:   9.0909090909091282e-10    9.0909090909091282e-10
> Overall NLP error.......:   9.0909090909091282e-10    9.0909090909091282e-10
> 
> 
> Number of objective function evaluations             = 33
> Number of objective gradient evaluations             = 30
> Number of equality constraint evaluations            = 33
> Number of inequality constraint evaluations          = 0
> Number of equality constraint Jacobian evaluations   = 30
> Number of inequality constraint Jacobian evaluations = 0
> Number of Lagrangian Hessian evaluations             = 29
> Total CPU secs in IPOPT (w/o function evaluations)   =      0.318
> Total CPU secs in NLP function evaluations           =      0.011
> ```
> 
> The solution is higher-quality and takes a reasonable number of iterations.
> 
> There may be some work to do on appropriate variable scaling, either on this flowsheet or the associated units.

Thanks @bknueven .  Yes, after this improvement, further improvement on scaling is likely to come with unit refinement.

**Comment by adam-a-a** [comment]
@lbibl Were you able to check whether these mods would resolve any test failures on PR #699 ? Also, while I was probing this the other day, I noticed in the IPOPT output that `power_electrical_x` was hitting its bound and IPOPT was adjusted variable bounds during iteration. I also noticed in the ED1D model, you set the first element for `power_electrical_x` == 0. I haven't tried running your mods, so maybe the issue doesn't show up anymore. Nevertheless, wonder if you might consider doing
```
power_electrical_x.fix(0)
return Constraint.Skip
```
instead of what you have in https://github.com/watertap-org/watertap/blob/main/watertap/unit_models/electrodialysis_1D.py#L713-L714

**Comment by lbibl** [comment]
> @lbibl Were you able to check whether these mods would resolve any test failures on PR #699 ? Also, while I was probing this the other day, I noticed in the IPOPT output that `power_electrical_x` was hitting its bound and IPOPT was adjusted variable bounds during iteration. I also noticed in the ED1D model, you set the first element for `power_electrical_x` == 0. I haven't tried running your mods, so maybe the issue doesn't show up anymore. Nevertheless, wonder if you might consider doing
> 
> ```
> power_electrical_x.fix(0)
> return Constraint.Skip
> ```
> 
> instead of what you have in https://github.com/watertap-org/watertap/blob/main/watertap/unit_models/electrodialysis_1D.py#L713-L714

Thanks @adam-a-a .  I can only test if this resolve the fails in #699  by merging this PR into it because I do not have Win64.  That will essentially duplicate the commits so I recommend merging this PR now.  #699 is expected to have a few minor changes too. 

The `power_elec` at x=0 also caught my eye.   I am seeking improvement in the ongoing refinement.

**Review by adam-a-a — APPROVED** [review]
LGTM

**Pull request merged**